### DISCUSSION TITLE
ABB/Power-One/FIMER PVI SunSpec Unload failing

### DISCUSSION THREAD

**Discussion by woodrook** [alexdelprete/ha-abb-powerone-pvi-sunspec#163] [Support]
Every morning integration show the message "Failed to unload" and i need to restart HA to let the inverter integraton works again.

Hi Alex, every evening my spook integration unload your ABB integration when the inverter goes off. Spook integration reload ABB when the inverter restart in the morning

This happened for many moths untill few days ago.

Now, every morning ABB integration give the error "Failed to unload" and I have to reboot HA to make it work again

The problem could be the "unloading" process that recently doesn't work becaus debug log show this

*****************************************************************
2024-07-24 21:08:01.977 ERROR (MainThread) [homeassistant.config_entries] Error unloading entry ABB Inverter for abb_powerone_pvi_sunspec
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/config_entries.py", line 801, in async_unload
    result = await component.async_unload_entry(hass, self)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/abb_powerone_pvi_sunspec/__init__.py", line 146, in async_unload_entry
    hass.data[DOMAIN][config_entry.entry_id].update_listener()
    ~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^
KeyError: '68e100a5b589d468733b76141e936def'
*****************************************************************

If I manually run the Spook automation the failure appear....what could be the problem?

HA Core recently updated to 2024.7.3 on docker on synology with Sonof ZB dongle-P

Thank you

**Comment by alexdelprete** [comment]
Sorry, I totally missed your post.

Yes, I found out there was a bug on unload/reload integration. I fixed it in latest beta, you can test it.

Anyway, there's no need of reloading integration anymore since `v3.4.0`. The integration now uses HA DataUpdateCoordinator, that automatically restarts fetching data when the inverter is available in the morning.

Let me know...
